core/src/main/java/com/linecorp/armeria/server/file/HttpFileServiceBuilder.java

```diff
@@ -130,18 +142,84 @@ public HttpFileServiceBuilder autoIndex(boolean autoIndex) {
         return this;
     }
 
+    /**
+     * Returns the immutable additional {@link HttpHeaders} which will be set when building an
+     * {@link HttpResponse}.
+     */
+    private HttpHeaders headers() {
+        return headers != null ? headers.asImmutable() : HttpHeaders.EMPTY_HEADERS;
+    }
+
+    private HttpHeaders getOrCreateHeaders() {
+        if (headers == null) {
+            headers = HttpHeaders.of();
+        }
+        return headers;
+    }
+
+    /**
+     * Adds the specified HTTP header.
+     */
+    public HttpFileServiceBuilder addHeader(CharSequence name, Object value) {
+        requireNonNull(name, "name");
+        requireNonNull(value, "value");
+        getOrCreateHeaders().addObject(HttpHeaderNames.of(name), value);
+        return this;
+    }
+
+    /**
+     * Adds the specified HTTP headers.
+     */
+    public HttpFileServiceBuilder addHeaders(Headers<AsciiString, String, ?> headers) {
+        requireNonNull(headers, "headers");
+        getOrCreateHeaders().add(headers);
+        return this;
+    }
+
+    /**
+     * Sets the specified HTTP header.
+     */
+    public HttpFileServiceBuilder setHeader(CharSequence name, Object value) {
+        requireNonNull(name, "name");
+        requireNonNull(value, "value");
+        getOrCreateHeaders().setObject(HttpHeaderNames.of(name), value);
+        return this;
+    }
+
+    /**
+     * Sets the specified HTTP headers.
+     */
+    public HttpFileServiceBuilder setHeaders(Headers<AsciiString, String, ?> headers) {
+        requireNonNull(headers, "headers");
+        if (!headers.isEmpty()) {
+            getOrCreateHeaders().setAll(headers);
+        }
+        return this;
+    }
+
+    /**
+     * Sets the {@code "cache-control"} header. This method is a shortcut of:
+     * <pre>{@code
+     * builder.setHeader(HttpHeaderNames.CACHE_CONTROL, cacheControl);
+     * }</pre>
+     */
+    public HttpFileServiceBuilder cacheControl(CacheControl cacheControl) {
+        requireNonNull(cacheControl, "cacheControl");
+        return setHeader(HttpHeaderNames.CACHE_CONTROL, cacheControl);
```

Comment:
nit: How about adding `HttpHeaders.cacheContrl(cc)` as well? It would be useful because our user doesn't need to write the header name. What do you think?

Reply:
Sure. Let me add one. :-)

Reply:
I changed my mind not to add it because:

- we cannot add a getter even if I added the parser because we cannot determine which parser should be used. `ClientCacheControl.parse()` vs `ServerCacheControl.parse()`.
- it feels weird to have only a setter while all other methods have setters.
- we will eventually make a large change on `HttpHeaders` for immutability, so perhaps it's not a good idea to add more setters.

Reply:
Agreed, especially the last one. :-)